### PR TITLE
Pass docker.io creds to build config step

### PR DESCRIPTION
**Pull request by Sandeepyadav93**
With this patch, we pass docker.io creds to build config to avoid rate limiting.

### CODE REVIEW

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-keystone-operator-master-keystone-operator-build-deploy-kuttl

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-ref.yaml [hunk 1]
```diff
@@ -14,8 +14,11 @@ ref:
       default: "openstack-k8s-operators"
       documentation: Registry organization to push containers built by this step.
   credentials:
-  - mount_path: /secrets/docker
+  - mount_path: /secrets/rdoquay
     name: rdoquay-oko-ci-robot
     namespace: test-credentials
+  - mount_path: /secrets/docker
+    name: docker-io-oko-creds
+    namespace: test-credential
```
Comment (viroel) [inline ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-ref.yaml:22]: it should be: "test-credentials"

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh [hunk 2]
```diff
@@ -66,6 +66,7 @@ function build_push_operator_images {
 
   # Build and push operator image
   oc new-build --binary --strategy=docker --name ${OPERATOR} --to=${IMAGE_TAG_BASE}:${IMAGE_TAG} --push-secret=${REGISTRY_SECRET} --to-docker=true
+  oc set build-secret --pull bc/${OPERATOR} ${DOCKER_REGISTRY_SECRET}
```
Comment (viroel) [inline ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh:69]: we probably don't need in the next 2 builds, but shouldn't we add there too?

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-keystone-operator-master-keystone-operator-build-deploy-kuttl

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-ref.yaml (hunk 1 above)
Comment (Sandeepyadav93) [reply ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-ref.yaml]: fixed in the latest patchset.

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh (hunk 2 above)
Comment (Sandeepyadav93) [reply ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh:69]: Hey Douglas,

If we look at logs[1] we only pull golang from docker.io, bundle and index don't pull anything from docker.io. So, its not really needed in the bundle/index build.


[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37285/rehearse-37285-pull-ci-openstack-k8s-operators-keystone-operator-master-keystone-operator-build-deploy-kuttl/1635685969749348352/artifacts/keystone-operator-build-deploy-kuttl/openstack-k8s-operators-build/build-log.txt

**Comment by viroel** [comment]
/lgtm

thanks Sandeep, lets see if that help us with rate limiting issues.

**Comment by viroel** [comment]
/pj-rehearse ack

build step log has the code change:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37285/rehearse-37285-pull-ci-openstack-k8s-operators-keystone-operator-master-keystone-operator-build-deploy-kuttl/1635685969749348352/artifacts/keystone-operator-build-deploy-kuttl/openstack-k8s-operators-build/build-log.txt

**Pull request merged**